README.md
# <a name="title"></a> THIS IS A FORK
Forked from [busser-serverspec][https://github.com/test-kitchen/busser-serverspec@0.5.10]


# <a name="title"></a> Busser::RunnerPlugin::Serverspec

A Busser runner plugin for Serverspec for EOL Ruby versions < 2.3 

## <a name="installation"></a> Installation and Setup

Please read the Busser [plugin usage][plugin_usage] page for more details.

## <a name="usage"></a> Usage

Please put test files into [COOKBOOK]/test/integration/[SUITES]/serverspec_eol/

```cookbook
`-- test
    `-- integration
        `-- default
            `-- serverspec_eol
                |-- Gemfile
                |-- localhost
                |   `-- httpd_spec.rb
                `-- spec_helper.rb
```

`Gemfile` is optional. You can specify installing Serverspec version and install gems you need.


CHANGELOG.md
# You should use Kitchen::Verifier::Shell + Serverspec

## 0.5.12 2019-03-22

* Fixes bundle install of serverspec-eol Gemfile on `kitchen verify`

## 0.5.11 2019-02-28

* Install bundler 1.17.3 for Ruby versions < 2.3
* fork and rename to busser-serverspec_eol
  
## 0.5.10 2016-08-12

* use container-based test at travis-ci.
* Make rake and rspec/core runtime deps
  merge pull-req by John Keiser <john@johnkeiser.com>, thanks.
* Pin versions of aruba and cucumber
  inspired by test-kitchen/busser-cucumber@cd3e6b3 by Peter Gallagher, thanks.

## 0.5.9 2016-01-22

* Fix frozen string issue when checking Ruby version
  merge pull-req by Joe Friedl <jfriedl@digitalocean.com>, thanks.

## 0.5.8 / 2016-01-22, 0.6.0 / 2016-01-18 (yanked 2016-02-22)

* Install net-ssh < 2.10 for ruby < 2.0.
  merge pull-req by Bill Ruddock <bill.ruddock@gmail.com>, thanks.

## 0.5.7 / 2015-06-01

* use RbConfig instead of obsolete and deprecated Config, closes #29
  suggested by Jörg Herzinger <joerg.herzinger@oiml.at>, thanks.

## 0.5.6 / 2015-04-27

* Run bundle with explicit path to ruby bin, closes #27
  merge pull-req by Simon Detheridge <simon@widgit.com>, thanks.

## 0.5.5 / 2015-04-02

* use File::PATH_SEPARATOR to join ENV PATH. closes #24
  suggested by Jacob McCann <jmccann.git@gmail.com>, thanks.

## 0.5.4 / 2015-03-30

* add the config bindir as well.
  merge pull-req by Noah Kantrowitz <noah@coderanger.net>. thanks.

## 0.5.3 / 2014-10-17

* make private working methods, closes #18

## 0.5.2 / 2014-10-16

* reset gems db

## 0.5.1 / 2014-10-16

* move Gemfile process from postinstall to test.

## 0.5.0 / 2014-10-16

* enable to specify serverspec version in Gemfile, closes #16
* add 'log_switch', '~> 0.3.0', ref: turboladen/tailor#160
* drop ruby 1.9.2 test, because mime-types 2.4.2 does not support ruby 1.9.2.

## 0.4.0 / 2014-10-06

* add --default-path for RSpec 3, closes #15
* bump up version number (0.3: load Gemfile, 0.4: serverspec v2).

## 0.2.8 / 2014-10-05

* update for serverspec v2, closes #13

## 0.2.7 / 2014-09-24

* support loading Gemfile. closes #12

## 0.2.6 / 2014-01-25

* add globbing pattern note for frequency asked test failure.
* update links for moving to test-kitchen organization.
* update copyright.

## 0.2.5 / 2013-11-14

* issue #5: Added --color and --format documentation rspec options
  merge pull-req by Kannan Manickam <kannan@rightscale.com>. thanks.

## 0.2.4 / 2013-11-01

* issue #4: Search all GEM_PATH values for `rspec` bin.
  merge pull-req by Seth Chisamore <schisamo@opscode.com>. thanks.

## 0.2.3 / 2013-07-10

* issue #2: remove tight dependency of serverspec. thanks to rteabeault.

## 0.2.2 / 2013-07-09

* issue #2: remove dependency rspec.

## 0.2.1 / 2013-06-18

* fix silly exit value.

## 0.2.0 / 2013-06-18

* issue #1: do not print full backtrace if serverspec test failure.

## 0.1.1 / 2013-06-06

* add cucumber test and fix related things

## 0.1.0 / 2013-05-22

* Initial release


lib/busser/runner_plugin/serverspec_eol.rb
# -*- encoding: utf-8 -*-
#
# Author:: HIGUCHI Daisuke (<d-higuchi@creationline.com>)
#
# Copyright (C) 2013-2014, HIGUCHI Daisuke
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

require 'busser/runner_plugin'
require 'rubygems/dependency_installer'

# A Busser runner plugin for Serverspec.
#
# @author HIGUCHI Daisuke <d-higuchi@creationline.com>
#
class Busser::RunnerPlugin::ServerspecEol < Busser::RunnerPlugin::Base
  postinstall do
    if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.3')
      install_gem('bundler', '~> 1.17.0')
    else
      install_gem('bundler')
    end
  end

  def test
    run_bundle_install
    install_serverspec

    runner = File.join(File.dirname(__FILE__), %w{.. serverspec runner.rb})
    run_ruby_script!("#{runner} #{suite_path('serverspec_eol').to_s}")
  end

  private

  def run_bundle_install
    # Referred from busser-shindo
    gemfile_path = File.join(suite_path, 'serverspec_eol', 'Gemfile')
    if File.exists?(gemfile_path)
      # Bundle install local completes quickly if the gems are already found
      # locally it fails if it needs to talk to the internet. The || below is
      # the fallback to the internet-enabled version. It's a speed optimization.
      banner('Bundle Installing..')
      ENV['PATH'] = [ENV['PATH'], Gem.bindir, RbConfig::CONFIG['bindir']].join(File::PATH_SEPARATOR)
      bundle_exec = "#{File.join(RbConfig::CONFIG['bindir'], 'ruby')} " +
        "#{File.join(Gem.bindir, 'bundle')} install --gemfile #{gemfile_path}"
      run("#{bundle_exec} --local || #{bundle_exec}")
    end
  end

  def install_serverspec
    Gem::Specification.reset
    if Array(Gem::Specification.find_all_by_name('serverspec')).size == 0
      if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.3')
        banner('Installing net-ssh < 2.10')
        install_gem('net-ssh', '< 2.10')
      end
      banner('Installing Serverspec..')
      spec = install_gem('serverspec')
      banner "serverspec installed (version #{spec.version})"
    end
  end
end
